### PR TITLE
[Improve][Issue-2494] forbid ldap user sign in with password

### LINKED ISSUE #2494: [Feature] improve ldap user management
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Streampark support `Sign in with LDAP` or `Sign in with password`. 

image

Suppose there is a LDAP user {name:user_a，password:123}, when this user login for the first time, Streampark will create a user {name:user_a，password:123}(here we ignore the encryption) in user system. This will lead two security problems:
- user_a are able to `Sign in with password` no matter what the status of LDAP service
- if user_a's password is changed in LDAP, before `Sign in with LDAP` again, user_a can still sign in with the old password.

So I propose to forbid LDAP user `Sign in with password`. Besides, the LDAP user is usually managed by some LDAP client, reset the password of a LDAP user in Streampark is unreasonable, we should hide the reset button for LDAP user in Streampark webui.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoulii**
## What changes were proposed in this pull request

Issue Number: close #2494 

## Verifying this change

- *Manually verified the change by testing locally.* 

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / **no**)

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/enums/LoginType.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.baomidou.mybatisplus.annotation.EnumValue;
+
+import java.util.Arrays;
+
+/** The user login type. */
+public enum LoginType {
+
+  /** sign in with password */
+  LOCAL(0),
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/enums/LoginType.java]: "PASSWORD" would be better?

Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/enums/LoginType.java]: I use PASSWORD first, then I find there is a `LoginTypeEnum` defined [useLogin.ts#L28](https://github.com/apache/incubator-streampark/blob/dev/streampark-console/streampark-console-webapp/src/views/base/login/useLogin.ts#L28), for consistency, LOCAL is used here.

Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/enums/LoginType.java]: > I use PASSWORD first, then I find there is a `LoginTypeEnum` defined [useLogin.ts#L28](https://github.com/apache/incubator-streampark/blob/dev/streampark-console/streampark-console-webapp/src/views/base/login/useLogin.ts#L28), for consistency, LOCAL is used here.

I still think "PASSWORD" would be better

Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/enums/LoginType.java]: done

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/PassportController.java [hunk 2]
```diff
@@ -114,4 +115,9 @@ private RestResponse login(String username, String password, User user) throws E
         userService.generateFrontendUserInfo(user, user.getLastTeamId(), jwtToken);
     return new RestResponse().data(userInfo);
   }
+
+  @PostMapping("types")
+  public RestResponse userTypes() {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/PassportController.java]: I don't think we need this method

Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/PassportController.java]: done. btw, refer to [UserController.java#L158](https://github.com/apache/incubator-streampark/blob/dev/streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/UserController.java#L158), we may alse need to refactor it.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java [hunk 3]
```diff
@@ -60,6 +61,9 @@ public User ldapAuthenticate(String username, String password) throws Exception
     User user = usersService.findByName(username);
 
     if (user != null) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java:78]: We can improve the code:
```
if (user == null || user.getLoginType() != LoginType.LDAP) {
  return null;
}
```
...

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java [hunk 4]
```diff
@@ -39,7 +40,7 @@ public class AuthenticatorImpl implements Authenticator {
   @Override
   public User authenticate(String username, String password) {
     User user = usersService.findByName(username);
-    if (user == null) {
+    if (user == null || user.getLoginType() == LoginType.LDAP) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java]: If we add another login type,  will a bug occur?
```
if (user == null || user.getLoginType() != LoginType.PASSWORD) {
      return null;
  }
```

Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java]: You're right, I will fix it.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java [hunk 5]
```diff
@@ -37,10 +39,29 @@ public class AuthenticatorImpl implements Authenticator {
   @Autowired private LdapService ldapService;
 
   @Override
-  public User authenticate(String username, String password) {
+  public User authenticate(String username, String password, String loginType) throws Exception {
+
+    LoginType loginTypeEnum;
+
+    try {
+      loginTypeEnum = LoginType.valueOf(loginType);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java]: oh no, We don't need to use `try`, 
```
 LoginType loginTypeEnum = LoginType.of(loginType);
if (loginTypeEnum == null) {
  // alert...
}

```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java [hunk 6]
```diff
@@ -37,10 +39,29 @@ public class AuthenticatorImpl implements Authenticator {
...
+    } else {
+      return ldapAuthenticate(username, password);
+    }
+  }
+
+  private User passwordAuthenticate(String username, String password) {
     User user = usersService.findByName(username);
-    if (user == null) {
-      return null;
+    if (user == null || user.getLoginType() != LoginType.PASSWORD) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java:58]: We don't need to use this judgment：
`user.getLoginType() != LoginType.PASSWORD`

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java [hunk 7]
```diff
@@ -60,6 +80,10 @@ public User ldapAuthenticate(String username, String password) throws Exception
     User user = usersService.findByName(username);
 
     if (user != null) {
+      if (user.getLoginType() != LoginType.LDAP) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/security/impl/AuthenticatorImpl.java:79]: We don't need to use this judgment

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**